src/main/kotlin/com/terraformation/backend/email/EmailNotificationService.kt

```diff
@@ -879,6 +881,35 @@ class EmailNotificationService(
     }
   }
 
+  @EventListener
+  fun on(event: RateLimitedT0DataAssignedEvent) {
+    if (event.monitoringPlots.flatMap { plot -> plot.speciesDensityChanges }.isEmpty()) {
+      // changes were reversed before the event was eventually refired
+      return
+    }
```

Comment:
Is this a sign we need a way for rate-limited events to delete themselves when they're combined?

Reply:
There are 2 other events that return early in `Email NotificationService`, though I don't think either are rate limited. I think that is one approach for sure. Perhaps we could change `combine` to return `T?` instead of `T`, and when null is returned after `combine`, we remove the event?

Reply:
Yeah, that's the same thought I had. Obviously not a topic for this PR, though (it's already big enough!)